src/main/java/com/mathworks/ci/MatlabBuild.java

```diff
@@ -85,26 +86,30 @@ default String getUniqueNameForRunnerFile() {
     
     // This method prepares the temp folder by coping all helper files in it.
     default void prepareTmpFldr(FilePath tmpFldr, String runnerScript) throws IOException, InterruptedException {
-        // Write MATLAB scratch file in temp folder.
-        FilePath scriptFile =
-                new FilePath(tmpFldr, getValidMatlabFileName(tmpFldr.getBaseName()) + ".m");
-        scriptFile.write(runnerScript, "UTF-8");
         // copy genscript package
-        copyFileInWorkspace(MatlabBuilderConstants.MATLAB_SCRIPT_GENERATOR,
-                MatlabBuilderConstants.MATLAB_SCRIPT_GENERATOR, tmpFldr);
-        FilePath zipFileLocation =
-                new FilePath(tmpFldr, MatlabBuilderConstants.MATLAB_SCRIPT_GENERATOR);
-
-        // Unzip the file in temp folder.
-        zipFileLocation.unzip(tmpFldr);
+    	 copyFileInWorkspace(MatlabBuilderConstants.MATLAB_SCRIPT_GENERATOR,
+                 MatlabBuilderConstants.MATLAB_SCRIPT_GENERATOR, tmpFldr);
+         FilePath zipFileLocation =
+                 new FilePath(tmpFldr, MatlabBuilderConstants.MATLAB_SCRIPT_GENERATOR);
+         runnerScript=replaceZipPlaceholder(runnerScript, zipFileLocation.getRemote());
+         
+      // Write MATLAB scratch file in temp folder.
+         FilePath scriptFile =
+                 new FilePath(tmpFldr, getValidMatlabFileName(tmpFldr.getBaseName()) + ".m");
+         scriptFile.write(runnerScript, "UTF-8");
     }

```

Comment:
Still something off with indentation in these files. Looks like a mixture of tabs and spaces are being used.